src/core/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,233 @@
+   * @default 'minutes'
+   */
+  precision?: 'hours' | 'minutes' | 'seconds';
+  /**
+   * Change step of the hours displayed.
+   * @default 1
+   */
+  hourStep?: number;
+  /**
+   * Change step of the minutes displayed.
+   * @default 1
+   */
+  minuteStep?: number;
+  /**
+   * Change step of the seconds displayed.
+   * @default 1
+   */
+  secondStep?: number;
+};
+
+/**
+ * Time picker component
+ * @example
+ * <TimePicker date={new Date()} onChange={(e) => console.log('New time value: ' + e)} />
+ */
+export const TimePicker = (props: TimePickerProps): JSX.Element => {
+  const {
+    date,
+    onChange,
+    use12Hours = false,
+    precision = 'minutes',
+    hourStep = 1,
+    minuteStep = 1,
+    secondStep = 1,
+  } = props;
+
+  useTheme();
+
+  const [selectedTime, setSelectedTime] = React.useState(date);
+
+  React.useEffect(() => {
+    setSelectedTime(date);
+  }, [date]);
+
+  const onHourClick = (hour: number) => {
+    const adjustedSelectedDay = selectedTime ?? new Date();
+    adjustedSelectedDay.setHours(hour);
+    updateCurrentTime(adjustedSelectedDay);
+  };
+
+  const onMinutesClick = (minutes: number) => {
+    const adjustedSelectedDay = selectedTime ?? new Date();
+    adjustedSelectedDay.setMinutes(minutes);
+    updateCurrentTime(adjustedSelectedDay);
+  };
+
+  const onSecondsClick = (seconds: number) => {
+    const adjustedSelectedDay = selectedTime ?? new Date();
+    adjustedSelectedDay.setSeconds(seconds);
+    updateCurrentTime(adjustedSelectedDay);
+  };
+
+  const updateCurrentTime = (time: Date) => {
+    setSelectedTime(time);
+    onChange?.(time);
+  };
+
+  // const handleTimeKeyDown = (
+  //   event: React.KeyboardEvent<HTMLLIElement>,
+  //   maxValue: number,
+  //   callback: (value: number) => void,
+  //   onClick: (value: number) => void,
+  //   currentValue?: number,
+  // ) => {
+  //   if (!focusedTime || currentValue == null) {
+  //     return;
+  //   }
+  //   switch (event.key) {
+  //     case 'ArrowDown':
+  //       if (currentValue + 1 > maxValue) {
+  //         break;
+  //       }
+  //       callback(currentValue + 1);
+  //       needFocus.current = true;
+  //       event.preventDefault();
+  //       break;
+  //     case 'ArrowUp':
+  //       if (currentValue - 1 < 0) {
+  //         break;
+  //       }
+  //       callback(currentValue - 1);
+  //       needFocus.current = true;
+  //       event.preventDefault();
+  //       break;
+  //     case 'Enter':
+  //     case ' ':
+  //     case 'Spacebar':
+  //       onClick(currentValue);
+  //       event.preventDefault();
+  //       break;
+  //   }
+  // };
+
+  const scrollIntoView = (ref: HTMLLIElement | null, isSame: boolean) => {
+    isSame && ref?.scrollIntoView({ block: 'nearest', inline: 'nearest' });
+  };
+
+  const getTimeTemplate = (
+    data: number[],
+    step: number,
+    isSame: (number: number, date: Date | undefined) => boolean,
+    onClick: (number: number) => void,
+  ) => {
```

Comment:
So this already has 4 params. What other ones do you think would be needed here to customize? If it's like 2-3 more props, I would say that's not too bad if they still reuse some code. If not, maybe we need 3 different template generators.

Reply:
@veekeys What other feedback do you need regarding this? I think this approach is pretty okay-ish

Reply:
I guess I will just try to implement something working from keyboard navigation and see, how it goes..

Reply:
Something is working!